Tests/DatadogTests/Datadog/Core/System/Time/DateCorrectionTests.swift

```diff
@@ -8,44 +8,30 @@ import XCTest
 @testable import Datadog
 
 private class ServerDateProviderMock: ServerDateProvider {
-    private(set) var synchronizedNTPPool: String? = nil
-    var offset: TimeInterval? = nil
+    private(set) var update: (TimeInterval) -> Void
+    private(set) var completion: (TimeInterval?) -> Void
 
-    init(using offset: TimeInterval? = nil) {
-        self.offset = offset
+    init() {
+        update = { _ in }
+        completion = { _ in }
     }
 
-    func synchronize(with pool: String, completion: @escaping (TimeInterval?) -> Void) {
-        synchronizedNTPPool = pool
-        completion(self.offset)
+    func synchronize(update: @escaping (TimeInterval) -> Void, completion:  @escaping (TimeInterval?) -> Void) {
+        self.update = update
+        self.completion = completion
     }
 }
 
 class DateCorrectorTests: XCTestCase {
-    func testWhenInitialized_itSynchronizesWithOneOfDatadogNTPServers() {
-        let serverDateProvider = ServerDateProviderMock()
-        let deviceDateProvider = SystemDateProvider()
-
-        var randomlyChosenServers: Set<String> = []
-
-        (0..<100).forEach { _ in
-            _ = DateCorrector(deviceDateProvider: deviceDateProvider, serverDateProvider: serverDateProvider)
-            randomlyChosenServers.insert(serverDateProvider.synchronizedNTPPool!)
-        }
-
-        let allAvailableServers = Set(DateCorrector.datadogNTPServers)
-        XCTAssertEqual(randomlyChosenServers, allAvailableServers, "Each time Datadog NTP server should be picked randomly.")
-    }
```

Comment:
This is no longer testable as the ntp pools have been moved to `NTPServerDateProvider`. We were only testing `Swift.Array.randomElement()`, so I think it is safe to remove. WDYT?

Reply:
Well, I disagree that it's not worth testing which pool we synchronise too (we need it to be DD pool for new skews in Distributed Tracing), but I agree that asserting on `DateCorrector.datadogNTPServers` didn't make much sense as it would be still green even after introducing a bug in the configuration (e.g. a typo).

Would be ideal to improve this test, not to remove it, WDYT @maxep?

Reply:
Yes, let me see what I can do 👍